### ISSUE TITLE
Add shard does not automatically creates schema

### ISSUE THREAD

**Issue by shinoi2** [Altinity/clickhouse-operator#227]
Root cause is `JOIN` logic changed in new version of clickhouse, but operator does not adaptation.

* https://clickhouse.yandex/docs/en/query_language/select/#select-join
* https://github.com/Altinity/clickhouse-operator/blob/master/pkg/model/schemer.go#L125
* clickhouse-operator log:
```
I1230 03:55:53.121183       1 schemer.go:61] clickhouse-opod.default.svc.cluster.local
I1230 03:55:53.172104       1 clickhouse.go:97] FAILED Query(http://clickhouse_operator:clickhouse_operator_password@c
lickhouse-opod.default.svc.cluster.local:8123/) Code: 48, Message: Old ANY INNER|RIGHT|FULL JOINs are disabled by defa
ult. Their logic would be changed. Old logic is many-to-one for all kinds of ANY JOINs. It's equil to apply distinct f
or right table keys. Default bahaviour is reserved for many-to-one LEFT JOIN, one-to-many RIGHT JOIN and one-to-one IN
NER JOIN. It would be equal to apply distinct for keys to right for SQL: SELECT DISTINCT 
        database AS name, 
        concat('CREATE DATABASE IF NOT EXISTS ', name) AS create_db_query
FROM 
(
        SELECT DISTINCT database
        FROM cluster('opod-cluster', system, tables)
        WHERE engine = 'Distributed'
        SETTINGS skip_unavailable_shards = 1
        UNION ALL
        SELECT DISTINCT extract(engine_full, 'Distributed\\([^,]+, *\'?([^,\']+)\'?, *[^,]+') AS shard_database
        FROM cluster('opod-cluster', system, tables) 
        WHERE engine = 'Distributed'
        SETTINGS skip_unavailable_shards = 1
) 
UNION ALL
SELECT DISTINCT 
        name, 
        replaceRegexpOne(create_table_query, 'CREATE (TABLE|VIEW|MATERIALIZED VIEW)', 'CREATE \\1 IF NOT EXISTS')
FROM 
(
        SELECT 
                database, 
                name,
                2 AS order
        FROM cluster('opod-cluster', system, tables)
        WHERE engine = 'Distributed'
        SETTINGS skip_unavailable_shards = 1
        UNION ALL
        SELECT 
                extract(engine_full, 'Distributed\\([^,]+, *\'?([^,\']+)\'?, *[^,]+') AS shard_database, 
                extract(engine_full, 'Distributed\\([^,]+, [^,]+, *\'?([^,\\\')]+)') AS shard_table,
                1 AS order
        FROM cluster('opod-cluster', system, tables)
        WHERE engine = 'Distributed'
        SETTINGS skip_unavailable_shards = 1
) 
ANY INNER JOIN (select distinct database, name, create_table_query from cluster('opod-cluster', system, tables) SETTIN
GS skip_unavailable_shards = 1) USING (database, name)
ORDER BY order
I1230 03:55:53.172148       1 retry.go:39] FAILED attempt 1 of 10, sleep 5 sec and retry: SELECT DISTINCT 
```

**Comment by alex-zaitsev** [comment]
Fixed in upcoming 0.8.0 already. You can take it from the branch or :dev tag on dockerhub.

**Comment by alex-zaitsev** [comment]
@shinoi2, https://github.com/Altinity/clickhouse-operator/releases/tag/0.8.0
